meinberlin/apps/dashboard2/forms.py

```diff
@@ -104,26 +49,102 @@ class Meta:
         }
 
 
-class ProjectBasicForm(forms.ModelForm):
+def _make_fields_required(fields, required):
+    """Set the required attributes on all fields who's key is in required."""
+    if required:
+        for name, field in fields:
+            if required == '__all__' or name in required:
+                field.required = True
+
+
+class ProjectDashboardForm(forms.ModelForm):
+    """
+    Base form for project related dashboard forms.
+
+    Sets fields to required if the project is published.
+    Intended to be used with ProjectFormComponent's.
+    """
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        if not self.instance.is_draft:
+            _make_fields_required(self.fields.items(),
+                                  self.get_required_fields())
+
+    @classmethod
+    def get_required_fields(cls):
+        meta = getattr(cls, 'Meta', None)
+        return getattr(meta, 'required', [])
+
+
+class ModuleDashboardForm(forms.ModelForm):
+    """
+    Base form for module related dashboard forms.
+
+    Sets fields to required if the project is published.
+    Intended to be used with ModuleFormComponent's.
+    """
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        if not self.instance.project.is_draft:
+            _make_fields_required(self.fields.items(),
+                                  self.get_required_fields())
+
+    @classmethod
+    def get_required_fields(cls):
+        meta = getattr(cls, 'Meta', None)
+        return getattr(meta, 'required', [])
+
+
+class ModuleDashboardFormSet(forms.BaseInlineFormSet):
+    """
+    Base form for module related dashboard formsets.
+
+    Sets fields to required if the project is published.
+    Intended to be used with ModuleFormSetComponent's.
+    """
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        if not self.instance.project.is_draft:
+            required_fields = self.get_required_fields()
+            for form in self.forms:
+                _make_fields_required(form.fields.items(),
+                                      required_fields)
+
+    @classmethod
+    def get_required_fields(cls):
+        meta = getattr(cls.form, 'Meta', None)
+        return getattr(meta, 'required', [])
+
+
+class ProjectBasicForm(ProjectDashboardForm):
 
     class Meta:
         model = project_models.Project
         fields = ['name', 'description', 'image', 'tile_image', 'is_archived',
                   'is_public']
+        required = '__all__'
```

Comment:
`required` is a very general name. Maybe change it to something like `required_for_project_publish`.